js/index.js

```diff
@@ -2,41 +2,85 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  
+
+const price = product.querySelector('.price span');
+const quantity = product.querySelector('.quantity input')
+
+const priceValue = price.innerText
+
+subtotal = priceValue * quantity.value
+
+console.log(priceValue)
+console.log(subtotal)
+
+
+const finalPrice = product.querySelector('.subtotal span')
+finalPrice.innerText = subtotal.toFixed(2)
 
-  //... your code goes here
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
+function calculateAll() {
+ 
   // ITERATION 2
-  //... your code goes here
+
+
+  const products = document.querySelectorAll('.product').forEach(product => {
+    updateSubtotal (product);
+
+  });
+
 
   // ITERATION 3
-  //... your code goes here
+  const eachTotal = document.querySelectorAll('.subtotal span');
+  let allTotal = 0;
+
+  for (let i = 0; i < eachTotal.length; i++) {
+    allTotal += parseInt(eachTotal[i].innerHTML);
+  }
+  document.querySelector('#total-value span').innerHTML = allTotal;
 }
```

Comment:
Good job for this one. Another way to do it is use the return from `updateSubtotal` to calculate the total sum. See below:

```
  let totalPrice = 0;
  products.forEach(function(product) { 
    totalPrice += updateSubtotal(product);
  })
```